Update the root readme (#1732)
Azure/azure-sdk-for-rust: README.md

# Azure SDK for Rust

> This repository is for the active development of the Azure SDK for Rust. For consumers of the SDK you can follow the links below to visit the documentation you are interested in

Reviewer: Punctuation:
```suggestion
This repository is for the active development of the Azure SDK for Rust. For consumers of the SDK you can follow the links below to visit the documentation you are interested in.
```

* [Overview of Azure SDK for Rust](https://docs.microsoft.com/azure/developer/rust/)
* [SDK Reference](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk)
* [Code Samples for Azure SDK for Rust](https://github.com/azure-samples/azure-sdk-for-rust-samples)
* [Azure REST API Docs](https://docs.microsoft.com/rest/api/)
* [General Azure Docs](https://docs.microsoft.com/azure)
* [Share your feedback to our Azure SDK](https://www.surveymonkey.com/r/FWPGFGG)

## Getting Started

To get started with a crate, see the README.md file located in the crate's project folder.  You can find these crate folders grouped by service in the `/sdk` directory.

## Crates available

> Each service can have both 'client' and 'management' crates. 'Client' crates are used to consume the service, whereas 'management' crates are used to configure and manage the service.

Reviewer: It may be first in a sentence, but the casing is specific. You can fix it with a colon.
```suggestion
Each service can have both 'client' and 'management' crates: 'client' crates are used to consume the service, whereas 'management' crates are used to configure and manage the service.
```

[All Azure SDK for Rust crates](https://crates.io/teams/github:azure:azure-sdk-publish-rust) are published on crates.io.

### Client modules

> Our client crates follow the [Azure Rust SDK guidelines](https://azure.github.io/azure-sdk/rust_introduction.html). These crates allow you to use, consume, and interact with existing resources, for example, uploading a blob. They also share a number of core functionalities including retries, logging, transport protocols, authentication protocols, etc. that can be found in the [azure_identity](https://crates.io/crates/azure_identity) module.

Reviewer: Consistency and punctuation:
```suggestion
Our crates follow the [Azure SDK for Rust guidelines](https://azure.github.io/azure-sdk/rust_introduction.html). These crates allow you to use, consume, and interact with existing resources; for example, uploading a blob. They also share a number of core functionalities including retries, logging, transport protocols, authentication protocols, etc. that can be found in the [azure_identity](https://crates.io/crates/azure_identity) crate.
```

> You can find the most up-to-date list of new modules on our [latest page](https://azure.github.io/azure-sdk/releases/latest/index.html#rust).

Reviewer: Translates better, if ever important:
```suggestion
You can find the most recent list of new crates on our [latest releases page](https://azure.github.io/azure-sdk/releases/latest/index.html#rust).
```

Also, stop calling them "modules". This isn't Go. :smile:

🚨 WARNING 🚨: This project is under active development. Be aware that large breaking changes will happen before 1.0 is reached.

### Historical releases

> This project is the successor to the `azure_sdk*` crates from [MindFlavor/AzureSDKForRust](https://github.com/MindFlavor/AzureSDKForRust). The crates have been renamed, so those older crates should be considered fully deprecated. See [history](HISTORY.md) for more details.

Reviewer 2: Do you also want to include a link to the legacy branch holding the unofficial Azure SDK Releases?

Reviewer: We haven't migrated yet and I don't think we should until we're ready to migrate `feature/track2` to `main`. People are still actively contributing to `main`. I think at that time this README should reflect that information, though, yes.

## Reporting security issues and security bugs

Security issues and bugs should be reported privately, via email, to the Microsoft Security Response Center (MSRC) <secure@microsoft.com>. You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we received your original message. Further information, including the MSRC PGP key, can be found in the [Security TechCenter](https://www.microsoft.com/msrc/faqs-report-an-issue).

## Need help?

> * File an issue via [Github Issues](https://github.com/Azure/azure-sdk-for-rust/issues)
> * Check [previous questions](https://stackoverflow.com/questions/tagged/azure+rust) or ask new ones on StackOverflow using `azure` and `rust` tags.

Reviewer: I think the order shouild be reversed. It's an unordered list, sure, but there's an implied order there. Also, there is a space in "Stack Overflow". It's a product name and should be referenced properly. GitHub also has a capital "G" and "H".

```suggestion
* Check [previous questions](https://stackoverflow.com/questions/tagged/azure+rust) or ask new ones on Stack Overflow using `azure` and `rust` tags.
* File an issue via [GitHub Issues](https://github.com/Azure/azure-sdk-for-rust/issues)
```

## Community

> ## Contribute

Reviewer: Empty heading.

We should put information that was here in a `CONTRIBUTING.md` at some point.
